### ISSUE TITLE
Could I make Hip-pop style rhythm on Creative-GAN? 

### ISSUE THREAD

**Issue by MCLYang** [naotokui/CreativeGAN-Rhythm#2]
Hi Dr. Tokui, 
I very like your idea and I am try the code on my local machine. I have read the paper and have some questions. Now I am focusing on generating single-style rhythm(for instance Hip-pop). Without modify Creative-GAN model architecture, could I achieve this objective by training Hip-pop music only? Would you recommend to try Genre-conditioned GAN or with Genre Ambiguity Loss? Any guidance will be appreciated.

**Comment by naotokui** [comment]
Sorry for my belated reply!  You can use Genre-Conditioned GAN for generating rhythm patterns in a certain style.  Or you can use the vanilla GAN model with your MIDI dataset containing hiphop rhythm patterns only.

**Comment by MCLYang** [comment]
Thank you for your response! I run your the notebook and very works well! I Implement the pytorch version of conditional GAN so I might en-scale the model. I carefully mapped your Keras code to pytorch and I have double checked for days. However,The Generator does not work as I expect. Could you please take a look of my code?
Here is link on google colab. You can run the code over there without downloading it: https://colab.research.google.com/drive/1hSGwu-Ie1q2RI4HeDbN5aAnli_xe_WZs?usp=sharing

To save your time, here is the evidence that I found might be useful:
1. I parallelly trained on Pytorch(My code) and Keras(Src code) and compare Loss each epoch. For Keras version, the average of g_loss for each epoch is rarely above 10. However, For Pytorch version the g_loss will go above 10 and even 20. I plot the result and found the generator always generate the sheets with one-hit at beginning but blank on every where else.  
Keras training result for first 10 epochs:

![image](https://user-images.githubusercontent.com/46518089/107113628-fd1b6e00-6814-11eb-89b5-c7d7544c1683.png)

Pytorch training result for first 10 epochs:
![image](https://user-images.githubusercontent.com/46518089/107113602-d52c0a80-6814-11eb-8cdb-cd5c7145778a.png)

2. To resolve the [1], I dive in LSTM block and I highly doubt Pytorch's LSTM modules are functioning differently than Keras. For a extreme case, when I replaced the LSTM by linear layers for pytorch, it still can learn some pasterns and even better LSTM, which means I might not properly configure LSTM for pytorch. Therefore, I hope you could confirm that if I use LSTM for pytorch correctly. Here is the docs for LSTM for pytorch: https://pytorch.org/docs/stable/generated/torch.nn.LSTM.html

I am looking forward to your reply! Thank you again.

**Comment by MCLYang** [comment]
NVM! I eventually solved the problem. For pytorch, we might need to manually initiate the hidden state and I add one linear layer after LSTM block then every thing works. Here is the latest result: https://colab.research.google.com/drive/1JOVz0n1jz-rSIkEDnCEm7omqATAwpuk5?usp=sharing
![image](https://user-images.githubusercontent.com/46518089/107129993-713e2c00-687e-11eb-9dfc-3ddd3c114016.png)

Please Let me know how do you think :)
